### ISSUE TITLE
Missing/Wrong requirements for TF2

### ISSUE THREAD

**Issue by Flamefire** [easybuilders/easybuild-easyconfigs#9306]
I ran into crashes and issues like https://github.com/tensorflow/tensorflow/issues/32319 using the current TF 2 EC and traced it back to wrong dependencies. Most notably it requires gast==0.2.2

I'm currently in the process of fixing them. For reference the dependencies required for TF2 are:

```
REQUIRED_PACKAGES = [
    'absl-py >= 0.7.0',
    'astor >= 0.6.0',
    'gast == 0.2.2',
    'google_pasta >= 0.1.6',
    'keras_applications >= 1.0.8',
    'keras_preprocessing >= 1.0.5',
    'numpy >= 1.16.0, < 2.0',
    'opt_einsum >= 2.3.2',
    'six >= 1.10.0',
    'protobuf >= 3.6.1',
    'tensorboard >= 2.0.0, < 2.1.0',
    'tensorflow_estimator >= 2.0.0, < 2.1.0',
    'termcolor >= 1.1.0',
    'wrapt >= 1.11.1',
    'grpcio >= 1.8.6',
    'wheel >= 0.26'
]
```

And current master (future):

```
REQUIRED_PACKAGES = [
    'absl-py >= 0.7.0',
    'astor >= 0.6.0',
    'gast == 0.2.2',
    'google_pasta >= 0.1.8',
    'keras_applications >= 1.0.8',
    'keras_preprocessing >= 1.1.0',
    'numpy >= 1.16.0, < 2.0',
    'opt_einsum >= 2.3.2',
    'protobuf >= 3.8.0',
    'tensorboard >= 2.0.0, < 2.1.0',
    'tensorflow_estimator >= 2.0.0, < 2.1.0',
    'termcolor >= 1.1.0',
    'wrapt >= 1.11.1',
    # python3 requires wheel 0.26
    'wheel >= 0.26;python_version>="3"',
    'six >= 1.12.0',
    'grpcio >= 1.8.6'
]
```

I'll create a PR on Monday with a updated and verified version.

Ccing @boegel 

Edit:
Using the current EC and issuing `pip install gast==0.2.2` yields:
```
Collecting gast==0.2.2
ERROR: tensorflow 2.0.0 requires google-pasta>=0.1.6, which is not installed.
ERROR: tensorflow 2.0.0 requires opt-einsum>=2.3.2, which is not installed.
```

Is there a "check" mode for pip to verify a packages dependencies which we could put into the EB?

**Comment by boegel** [comment]
Tagged for v4.1.0 so we don't overlook fixing this, thanks for opening the bug report @Flamefire!

The easiest way I've found to figure out which dependencies there are is to install the package with `pip install` to a temporary location, and then using a tool `pipdeptree` to print the dependency tree.

Something like this:

```
module load SciPy-bundle/2019.03-foss-2019a   # just to avoid that numpy/scipy is installed by pip
export XDG_CACHE_HOME=/tmp/$USER/pip_cache
pkg=gast==0.2.2
virtualenv --system-site-packages /tmp/$USER/$pkg
source /tmp/$USER/$pkg/bin/activate
pip install $pkg
pip install pipdeptree
pipdeptree -j -p $pkg > ${pkg}_deptree.json
deactivate
```
```

**Comment by boegel** [comment]
@Flamefire Is there an easy way to trigger that problem?

Maybe we could add another smoke test like [TensorFlow-2.x_mnist-test.py](https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.x_mnist-test.py)...

**Comment by Flamefire** [comment]
I have a kinda reduced example but not really done yet. It requires a large model and tensorflow_datasets. I'd rather have something like `pip verify-dependencies tensorflow` or so

**Comment by boegel** [comment]
Maybe `pip check`?

**Comment by boegel** [comment]
see https://pip.pypa.io/en/stable/reference/pip_check/

**Comment by boegel** [comment]
That seems to work, here's what I get from `pip check` when running it from the `tensorflow-2.0.0` unpacked source dir after loading `TensorFlow/2.0.0-foss-2019a-Python-3.7.2`:

```
$ pip check
tensorflow 2.0.0 requires google-pasta, which is not installed.
tensorflow 2.0.0 requires opt-einsum, which is not installed.
tensorflow 2.0.0 has requirement gast==0.2.2, but you have gast 0.3.2.
tensorboard 2.0.0 has requirement setuptools>=41.0.0, but you have setuptools 40.8.0.
$ echo $?
1
```

I'll look into adding support for running `pip check` in `PythonPackage` (we'll have to make that opt-in for now).

**Comment by boegel** [comment]
fixed in #9338, thanks @Flamefire!
